provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -123,8 +128,8 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
    * 2) return the extension of part.external_reference, if recognized
    * 3) throw IllegalArgumentException otherwise
    */
-  private def getFormat(part: JobPart): String =
-    part.format.orElse(PartialFunction.condOpt[String, String](part.external_reference.split("\\.").last) {
+  private def getFormat(part: JobPart): String = {
+    val ret = part.format.orElse(PartialFunction.condOpt[String, String](part.external_reference.split("\\.").last) {
       _ match {
```

Comment:
I condensed the partial function, but checkFormat needs to stay where it is since you still want to check the format if it's explicitly set and we don't try to derive it.